src/main/scala/com/databricks/spark/sql/perf/tpcds/Tables.scala

```diff
@@ -85,14 +79,70 @@ class Tables(sqlContext: SQLContext, dsdgenDir: String, scaleFactor: Int) extend
 
       val convertedData = {
         val columns = schema.fields.map { f =>
-          val columnName = new ColumnName(f.name)
-          columnName.cast(f.dataType).as(f.name)
+          col(f.name).cast(f.dataType).as(f.name)
         }
         stringData.select(columns: _*)
       }
 
       convertedData
     }
+
+    def genData(location: String, format: String, overwrite: Boolean): Unit = {
+      val mode = if (overwrite) SaveMode.Overwrite else SaveMode.Ignore
+
+      val withFormatAndMode = df.write.format(format).mode(mode)
+      val withPartitionColumns = if (partitionColumns.isEmpty) {
```

Comment:
The builder I think is mutable, so you can just call the methods on it instead of saving it to a new val.